GiEnJul/Controllers/RecipientController.cs

```diff
@@ -26,6 +26,11 @@ public RecipientController(IRecipientRepository recipientRepository, IPersonRepo
         public async Task<ActionResult<Entities.Recipient>> PostAsync([FromBody] Models.Recipient recipient)
         {
             _log.Debug("Adding recipient object: {@recipient}", recipient);
+            if (string.IsNullOrEmpty(recipient.Location)) {
+                var e = new Exception("Cannot add recipient with no location.");
+                _log.Error("Exception occurred while trying to add recipient: {@0}. \n{@1}", recipient, e);
+                throw e;
```

Comment:
```suggestion
                _log.Error("Exception occurred while trying to add recipient: {@0}", recipient);
                throw new ArgumentNullException("Location", "Cannot add recipient with no location");
```

Not sure if I used everything correctly here, maybe test this locally first

Reply:
Moved the null value check into the controller.